packages/admin/admin/src/form/ChipSelect.tsx

```diff
@@ -0,0 +1,187 @@
+            min-height: unset !important;
+        }
+    `,
+);
+
+export interface ChipSelectProps<T = string>
+    extends ThemedComponentBaseProps<{
+            root: "div";
+            inputRoot: "div";
+            select: typeof Select;
+            menuItem: typeof MenuItem;
+            inputBase: typeof MuiInputBase;
+        }>,
+        Omit<ChipProps, "children" | "onChange"> {
+    children?: React.ReactNode;
+    getOptionLabel?: (option: T) => string;
+    getOptionValue?: (option: T) => string;
+    options?: T[];
+    selectedOption?: T;
+    onChange: (event: SelectChangeEvent) => void;
+    fullWidth?: boolean;
+}
+
+const ChipInput = ({
+    chipProps,
+    inputBaseProps,
+    inputRootProps,
+    ...restProps
+}: InputBaseProps & {
+    chipProps?: Omit<ChipProps, "children">;
+    inputBaseProps?: InputBaseProps;
+    inputRootProps?: React.ComponentPropsWithoutRef<"div">;
+}) => {
+    return (
+        <ChipInputRoot {...inputRootProps}>
+            <Chip icon={<ChevronDown />} label={restProps.value?.toString().length ? restProps.value : ""} variant="filled" {...chipProps} />
+            <InputBase
+                sx={{
+                    position: "absolute",
+                    top: 0,
+                    left: 0,
+                    right: 0,
+                    bottom: 0,
+                    opacity: 0,
+                }}
+                {...inputBaseProps}
+                {...restProps}
+            />
+        </ChipInputRoot>
+    );
+};
+
+export function ChipSelect<T = string>(inProps: ChipSelectProps<T>) {
+    const {
+        options,
+        getOptionLabel = (option: T) => {
+            if (typeof option === "object" && option !== null) {
+                if ((option as any).label) return String((option as any).label);
+                if ((option as any).id) return String((option as any).id);
+                if ((option as any).value) return String((option as any).value);
+                return JSON.stringify(option);
+            } else {
+                return String(option);
+            }
+        },
+        getOptionValue = (option: T) => {
+            if (typeof option === "object" && option !== null) {
+                if ((option as any).value) return String((option as any).value);
+                if ((option as any).id) return String((option as any).id);
+                return JSON.stringify(option);
+            } else {
+                return String(option);
+            }
+        },
+        selectedOption,
+        children,
+        slotProps,
+        onChange,
+        fullWidth = false,
+        className,
+        ...chipProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminChipSelect",
+    });
+
+    const ownerState = { fullWidth };
+
+    return (
+        <Root {...slotProps?.root} ownerState={ownerState}>
+            <Select
+                {...slotProps?.select}
+                value={selectedOption || ""}
```

Comment:
Is having a default value here a good idea?

Reply:
Mui recommends doing that as can be seen here: https://arc.net/l/quote/jbfehneo